File: Code/20_Outros/Temporizacao/Ex1.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
	struct tm strtime;
	time_t timeoftheday;

	// Hora atual do sistema
	timeoftheday = time(NULL);
	// Apresenta data e hora em formato padrão
	printf("Data atual:\n   %s",
		ctime(&timeoftheday));

	puts("Setando estrutura para\n"
		"   04/02/2010, 03:30:38");
	// Anos desde 1900
	strtime.tm_year = 2010-1900;
	// Meses desde janeiro, 0-11
	strtime.tm_mon = 1;
	// Dias, 1-31
	strtime.tm_mday = 4;
	// Horas desde meia-noite, 0-23
	strtime.tm_hour = 02;
	// Minutos, 0-59
	strtime.tm_min = 30;
	// Segundos, 0-59
	strtime.tm_sec = 38;
	// Horario de verao?
	strtime.tm_isdst = 0;
	timeoftheday = mktime(&strtime);
	printf("Tempo retornado:\n   %s",
		ctime(&timeoftheday));
	printf("Conferindo resultado "
		"com programa 'cal':\n\n");
	system("cal 2 2010");
	return 0;
}

File: 4_Dicas/4.5_Temporização/Python/Ex1.py
import time, os

timeoftheday = time.time()
print("Data atual:\n   " + time.ctime(timeoftheday))

dias = ["Segunda","Terça", "Quarta", "Quinta","Sexta", "Sábado","Domingo"]
tm_s = time.localtime(timeoftheday)
print("Dia da semana: " + dias[tm_s.tm_wday])

print("Setando estrutura para\n   04/02/2010, 03:30:38")
# Ano, mês, dia do mês, hora-1, minutos, segundos, dia da semana, dia do ano, horário de verão
strtime = time.struct_time([2010,2,4,2,30,38,0,0,0])
print("Tempo retornado:\n   " + time.ctime(time.mktime(strtime)))
print("Conferindo resultado com programa 'cal':")
os.system("cal 2 2010")